lab_5_scrapper/scrapper.py

```diff
@@ -140,12 +262,64 @@ def get_search_urls(self) -> list:
         Returns:
             list: seed_urls param
         """
+        return self.config.get_seed_urls()
 
 
 # 10
 # 4, 6, 8, 10
 
 
+class CrawlerRecursive(Crawler):
+    def __init__(self, config: Config) -> None:
+        super().__init__(config)
+        self.base_url = self.config.get_seed_urls()[0]
+        self.visited_urls = [self.base_url]
+        self.page_url = self.base_url
+        self.urls = []
+
+    def get_info(self) -> None:
+        if (ASSETS_PATH.parent / 'recursive_crawler.json').exists():
+            print('yes')
+            with open(ASSETS_PATH.parent / 'recursive_crawler.json', 'r', encoding='utf-8') as infile:
+                data = json.load(infile)
+
+            self.visited_urls = data['visited_urls']
+            self.page_url = data['page_url']
+            self.urls = data['article_urls']
+
+    def save_info(self) -> None:
+        data = {
+            'visited_urls': self.visited_urls,
+            'page_url': self.page_url,
+            'article_urls': self.urls
+        }
+        with open(ASSETS_PATH.parent / 'recursive_crawler.json', 'w',
+                  encoding='utf-8') as file:
+            json.dump(data, file, indent=4)
+
+    def find_articles(self) -> None:
+        for item in range(4):
+            url = self.page_url
+            response = make_request(url, self.config)
+            if not response.ok:
+                continue
+            article_bs = BeautifulSoup(response.content, "html.parser")
+            links = article_bs.find_all('a', class_='listing-preview__content')
+            urls = []
+            for link in links:
+                urls.append(link['href'])
+            for url in urls:
+                if len(self.urls) == self.config.get_num_articles():
+                    self.save_info()
```

Comment:
save only on collecting all?? what if I need 100k urls? how probable that your machine never sleeps or crashes until you reach it?